001 bakery

fabricematrat opened the pull request.

Commit:
Update unit count on subordinates

anthonydillon committed:
Create the add relation icon and animate it

anthonydillon committed:
Positioning and animating the add-realtion icon

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -35,260 +35,267 @@ YUI.add('juju-env-bakery', function(Y) {
+     @param {Function} successCallback Called when the api request completes
+            successfully.
+     @param {Function} failureCallback Called when the api request fails
+            with a response of >= 400 except 401 and a WW-Authenticate
```
s/WW/WWW/

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -35,260 +35,267 @@ YUI.add('juju-env-bakery', function(Y) {
+     Initialize.
+
+     @method initializer
+     @param {Object} cfg Application configuration data.
```
Please document what is expected to be in this. How is the caller meant to know what to pass in?

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -395,27 +415,35 @@ YUI.add('juju-env-bakery', function(Y) {
       check(plain.Condition);
-      var rootKey = sjcl.codec.base64.toBits(plain.RootKey);
-      
+      var rootKey = [];
```
This seems wrong to me. newMacaroon says that its first argument must be an sjcl bitArray, but this is making a normal array. If this works, it's only coincidental and I don't think we should rely on that. As I suggested before, the right solution here is make the macaroon library use UInt8Arrays in its public interface.

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -395,27 +415,35 @@ YUI.add('juju-env-bakery', function(Y) {
-      @return {String} Macaroons that was set in local cookie.
-    */
+     @method _getMacaroon
+     @return {String} Macaroons that was set in local cookie.
```
s/Macaroons/Macaroon/

(isn't this assuming there can only be one macaroon in the local cookie store? is that correct?)

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -356,23 +363,36 @@ YUI.add('juju-env-bakery', function(Y) {
-      m.addThirdPartyCaveat(toSJCLbits(rootKey), caveatId, location);
+
+      var str = nacl.util.encodeBase64(rootKey);
+      var bytes = [];
```
The same applies here as to the rootKey parameter to newMacaroon below.

rogpeppe commented on jujugui/static/gui/src/app/store/env/bakery.js:
```diff
@@ -301,14 +308,15 @@ YUI.add('juju-env-bakery', function(Y) {
+     @method _interact
+     @param {Function} The callback function to be sent in case of
+            successful authentication.
+     @param {Function} The callback function failure in case of
```
This doc doesn't mention what the third parameter (actually the most important one) is.

rogpeppe commented:
Looks reasonable except that we shouldn't be passing non-bitArrays into macaroon. This might appear to work, but it doesn't actually work - [1, 2] does not represent the same bits as sjcl.codec.hex.toBits("0102") when treated as a bitArray, so this logic will be creating corrupted macaroons. Let's fix macaroon.js to use Uint8Arrays.

rogpeppe committed:
Add third party caveats and tests.

kadams54 committed:
Add babel preprocessor.

huwshimi committed:
Fix navigating back to expose on delete.

huwshimi committed:
Fixed passing of new props to the deployment components.

kadams54 committed:
Add entity content and description.

huwshimi committed:
Update the id of the unit (doesn't properly fix the issue).

huwshimi committed:
Added beginnings of the relations components

jujugui committed:
Merge pull request #905 from huwshimi/add-expose

Add expose

Add the expose view to the inspector.

jujugui committed:
Merge pull request #909 from huwshimi/fix-remove-unit-update

Fix deployment props update

Fixed passing of new props to the deployment components.

huwshimi committed:
Add ability to desploy bundles.

huwshimi committed:
Added test for deploying a bundle.

kadams54 committed:
Remove dummy content.

jujugui committed:
Merge pull request #910 from huwshimi/bundle-add-to-canvas

Bundle add to canvas

Added functionality to add a bundle to the canvas.

Commit:
Updated demo env switcher to new toggle UI

Commit:
Updated the env list to the new UI and functionality

Commit:
Update subordinate unit count per relations.

Missed .only

Forgotten semicolons

responding to reviews

Softwrap hid a long line

Remove extra comma

Added tests for the new subordinate unit list functionalities

Added tests for the new subordinate unit list functionalities

Clicking units in the subordinate unit list now show that unit and clicking 'back' now goes back to the subordinate instead of the host

huwshimi committed:
Merge branch 'entity-content' of https://github.com/kadams54/juju-gui into kadams54-entity-content

huwshimi committed:
Cleanups and moved config options to own component.

huwshimi committed:
Added the readme.

huwshimi committed:
Display correct info for charms/bundles.

huwshimi committed:
Added display of bundle diagrams.

huwshimi committed:
Improved entity details tests. Added test for displaying a diagram.

huwshimi committed:
Add tests for content.

Commit:
Update env commands to support the JEM endpoints as well

Commit:
Enable button; remove old service menu

Commit:
Generate a proper socket url when using JEM to list environments

jujugui committed:
Merge pull request #913 from huwshimi/kadams54-entity-content

Entity content

Display content for the bundles and charms.

Commit:
Switch to using d3 to cancel selection

huwshimi committed:
Display the onboarding.

Commit:
Help with circle smoothing

huwshimi committed:
Display the relations.

frankban committed:
Add support for creating new models on openstack controllers.

frankban committed:
Improve docstring.

huwshimi committed:
Add tests for the realtions components.

huwshimi committed:
Correctly get existing property.

Commit:
Clean up model after each test.

kadams54 committed:
Implement the spinner as a standalone component.

kadams54 committed:
Fix failing tests.

jujugui committed:
Merge pull request #919 from kadams54/fix-entity-extension-tests

Fix entity extension tests.

Commit:
Updated switchEnv to accept user credentials

jujugui committed:
Merge pull request #917 from bac/kyle-test-fixes

Minimal test updates.

jujugui committed:
Merge pull request #915 from frankban/openstack-controller-support

Add support for creating new models on openstack controllers.

frankban committed:
Enable support for using Juju to get bundle changes.

Also fall back to GUI server endpoints if Juju bundle
deployment is not available.

jujugui committed:
Merge pull request #920 from frankban/core-bundle-changes

Enable support for using Juju to get bundle changes.

Also fall back to GUI server endpoints if Juju bundle
deployment is not available.

jujugui committed:
Merge pull request #914 from makyo/add-relation-button

Add relation button

Commit:
When creating a new environment wait until the envlist updates then switch

Commit:
UI Updates to make the demo UI cleaner

huwshimi committed:
Updated nav to the change version view.

frankban committed:
Fix bundle deployment in sandbox mode.

Commit:
Created tests for the environment switcher list component:

jujugui committed:
Merge pull request #922 from frankban/get-bundle-changes-sandbox

Fix bundle deployment in sandbox mode.

Commit:
Created tests for the env switcher component

Commit:
If you try and switch environments in sandbox mode it should silently fail

huwshimi committed:
Added Change version components.

huwshimi committed:
Show the versions and navigate to the charm details.

Commit:
Removed unused argument

huwshimi committed:
Add setting charms to new versions.

huwshimi committed:
Only show Change version for a committed service.

huwshimi committed:
Test the change version components.

kadams54 committed:
Finish setting up karma properly.

huwshimi committed:
Fix name lengths.

huwshimi committed:
Correctly display a message if there are no other versions. Fix class names.

huwshimi committed:
Fixed escape with react flag.

kadams54 committed:
Simplify test.

frankban committed:
wip

kadams54 committed:
Add dependency on spinner.

jujugui committed:
Merge pull request #926 from kadams54/spinner

Create a standalone Spinner component.

frankban committed:
use proper host-port

frankban committed:
wip

frankban committed:
return

frankban committed:
set up steate earlier

frankban committed:
ready at the end

Commit:
Update the environment list when opening the env listing and after getting a successful listing from JEM

huwshimi committed:
Fix object prop checking. Added a comment.

Commit:
Move the env name generation outside of the jsx

huwshimi committed:
Fix module conflict for spinner.

jujugui committed:
Merge pull request #927 from huwshimi/fix-spinner

Fix spinner

Fix module conflict for spinner.

jujugui committed:
Merge pull request #924 from huwshimi/fix-escape

Fixed escape with react flag.

Don't try and close the old deployer bar when pressing esc with the react flag.

Commit:
Use the GUI user auth instead of the env auth for display user name

Commit:
Use the GUI user auth instead of the env auth for display user name

Commit:
Set credentials on app load

Commit:
Only update the machine view title if it is open

huwshimi committed:
Fix long service names. Fix readme line height.

huwshimi committed:
Fade in config fields.

huwshimi committed:
Show the version details instead of linking to jujucharms.com.

huwshimi committed:
Update tests for long service names.

huwshimi committed:
Fix error when adding duplicate services.

huwshimi committed:
Don't try and display the readme if it has been unmounted.

huwshimi committed:
Update the model and the ecs record.

frankban committed:
First workaround.

frankban committed:
Proper fix with actually closing the connection.

frankban committed:
Improve slow connection handling. Rename websocket destroy to disconnect.

hatched committed:
Merge pull request #6 from frankban/plan-a

Fix WebSocket connection issues.

Commit:
change innerText to textContent in app.js

kadams54 committed:
Remove unecessary function param.

kadams54 committed:
Use a plain vanilla loop.

kadams54 committed:
Just pass in the whole option object.

Commit:
Fixed tests from the env switcher updates and added new test to make sure it loads the env list on open

Commit:
Fixed failing tests from app init changes

kadams54 committed:
Refactored makeEntity and fixed tests.

Commit:
Fixed the issue in sandbox mode where we were passing the incorrect credentials from the server

kadams54 committed:
Add propTypes.

kadams54 committed:
Clean up React warnings.

Commit:
Updated pyramid tests for sandbox login

huwshimi committed:
Fixed lint.

Commit:
Updated credentials for selenium tests

Commit:
run tests

jujugui committed:
Merge pull request #921 from hatched/real-env-switcher

Create the real environment switcher

Create an environment switcher which supports both JES and JEM

jujugui committed:
Merge pull request #916 from huwshimi/add-relations

Add relations

Add the relations view to the inspector.

Commit:
Add unload event listener

huwshimi committed:
Fixed dropping bundles outside the onboarding.

huwshimi committed:
Post rebase updates.

huwshimi committed:
Spaces between css class names.

huwshimi committed:
Spaces between class names.

huwshimi committed:
Cancel readme request when unmounting the component.

huwshimi committed:
Skip spurious test failure.

huwshimi committed:
Skip spurious test failure.

huwshimi committed:
Fixed lint and syntax.

huwshimi committed:
Fixed lint.

huwshimi committed:
Fixed lint.

Commit:
Bump revision number

jujugui committed:
Merge pull request #923 from huwshimi/add-change-version

Add change version

Added the UI for changing the version of a service.

Commit:
Semicolons!

jujugui committed:
Merge pull request #932 from makyo/logout-delete-macaroons

Add unload event listener to wipe macaroons

jujugui committed:
Merge pull request #931 from huwshimi/drop-bundle

Drop bundles

Fixed dropping bundles outside the onboarding.

huwshimi committed:
Update the id map for the units.

huwshimi committed:
Update the idMaps.

huwshimi committed:
Moved unit id updating to a single method.

huwshimi committed:
Increment duplicate services names as they're added.

huwshimi committed:
Added comment and fix doc string.

huwshimi committed:
Include download results in the charm header.

jujugui committed:
Merge pull request #933 from huwshimi/update-uncommitted-unit-ids

Update uncommitted unit ids

Update uncommitted units ids when the service is committed.

jujugui committed:
Merge pull request #929 from huwshimi/ui-fixes

UI fixes

Fixed readme loading if the details have been closed. Fixed key error when adding duplicate services. Fade in config fields. Fixed readme line height. Fixed long service names in the service list.

frankban committed:
Create model in controller: support MAAS provider.

frankban committed:
Improve handling of relation-broken errors.

frankban committed:
Bump version up to 0.1.2.

jujugui committed:
Merge pull request #934 from frankban/create-env-maas

Support MAAS controllers when creating new models.

Also improve handling of relation-broken errors.

kadams54 committed:
Backwards-compatible support for CI Karma runs.

Commit:
Create new target to speed up babel builds

Commit:
Add node_modules dependency

kadams54 committed:
Lint fix.

kadams54 committed:
Fix null option issue.

kadams54 committed:
Add test for no options.

kadams54 committed:
Update function call.

jujugui committed:
Merge pull request #930 from kadams54/entity-content-cleanup

Entity content code cleanup.

Address the comments from https://github.com/juju/juju-gui/pull/913.

jcsackett committed:
Bump version.

jujugui committed:
Merge pull request #938 from jaycee/0.1.3

Bump version.

jujugui committed:
Merge pull request #936 from hatched/faster-babel

Create new target to speed up babel builds

To stop babel from starting a new instance for every file it's building on a fresh build use the new `fast-babel` target.

huwshimi committed:
Group erroring units by type.

huwshimi committed:
Added a spinner when the versions are loading. New spinner style. Use spinner for app loading.

kadams54 committed:
Add requested comments.

jujugui committed:
Merge pull request #935 from kadams54/good-karma

Optimize Karma for dev work.

This branch streamlines what's needed to run Karma in a dev environment, meaning we can iterate much faster through the dev/test cycle. It also makes it easier to run the Karma tests within IDEs, using whatever mechanisms they have for building.

jujugui committed:
Merge pull request #939 from huwshimi/group-errors

Group errors

Group erroring units by type.

Commit:
Upgraded react to 0.14.0 and react dependencies

huwshimi committed:
Removed spinner code.

jujugui committed:
Merge pull request #940 from huwshimi/versions-spinner

Versions spinner

Added a spinner when the versions are loading. New spinner style. Use the new spinner for app loading.

huwshimi committed:
Remove old deployer bar, enable new deployment components.

huwshimi committed:
Remove the old environment header. Enable the new component.

huwshimi committed:
Remove CSS import.

Commit:
jujulib doesn't select state server when creating an environment.

Commit:
env-switcher finds and supplies the state server.

Commit:
env-switcher uses the state server name as the name of the template to support multi-controller situations.

Commit:
Upgraded react to 0.14.0 and react dependencies

Commit:
sed replace the shallowRender code in the build react code to fix regression in 0.14

Commit:
Update React.render to ReactDOM.render

Commit:
Updated failing tests from the 0.14 update

Commit:
Use bumpversion. Next dist release will be 1.9.0

Commit:
Fixed makefile

Commit:
updated scale-service component to use refs instead of DOM queries

Commit:
Updated the header search and inspector config to react 0.14

Commit:
Updated test files to use ReactDOM.findDOMNode()

jujugui committed:
Merge pull request #944 from bac/use-bumpversion

Use bumpversion.

Also use the wheels/generic directory for download-cache.

Commit:
Added the react-dom script to the old mocha test index

Commit:
Added comment about React dist bugfix

jujugui committed:
Merge pull request #942 from huwshimi/remove-old-env-header

Remove old env header

Remove the old environment header. Enable the new component.

Commit:
Updated missed React to ReactDOM change. Fixed issue where a YUI event handler was too grabby

jujugui committed:
Merge pull request #941 from huwshimi/remove-old-deployer-bar

Remove old deployer bar

Remove old deployer bar, enable new deployment components.

Commit:
Fixed issue where a YUI event handler was trying to access methods on a react node

Commit:
Added comment as to why we don't use getDOMNode in the keypress handler

Commit:
Skipped test for code which is no longer being used and will be removed

jujugui committed:
Merge pull request #945 from hatched/upgrade-react

Upgrade React to 0.14

The latest release of React has been out for a while now, it brings a few nice changes that we can take advantage of so this branch makes the necessary modifications to upgrade React to 0.14

huwshimi committed:
Remove remaining react flags.

huwshimi committed:
Update tests to work with react flag removal.

huwshimi committed:
Improve tests.

Commit:
Added comment requested in review.

Commit:
Change name of server/template in tests per request in review.

Commit:
Updated environment switcher styling

Commit:
Pass the custom env name, if it exists to JEM or JES

Commit:
Updated tests for the new env name support

huwshimi committed:
Updated tests to work with flag removal

huwshimi committed:
Removed old test.

huwshimi committed:
Removed non react flag code.

jujugui committed:
Merge pull request #943 from jaycee/better-new-environment-assumptions

Use the state-server's name for the template when creating environments.

* state server selecting logic is now part of env switcher
* jujulib now uses whatever state server name it is given

Commit:
Fixed styling so that the inputs would display values in firefox

jujugui committed:
Merge pull request #947 from hatched/specify-env-name

Ability to specify a custom env name

As an interim feature we allow the user to specify their new environments name prior to creation if they don't want the auto-incremented name.

Commit:
Address comments

jujugui committed:
Merge pull request #946 from huwshimi/remove-inspector-flag

Remove inspector flag

Remove the remaining flags for the inspector, charmbrowser and env switcher. There will be a followup to actually remove the old code.

huwshimi committed:
Remove inspector, subapp etc.

huwshimi committed:
Removed remaining views and templates.

huwshimi committed:
Update and remove tests.

huwshimi committed:
Removed old icons.

huwshimi committed:
Removed unused utils.

huwshimi committed:
Removed old docs.

huwshimi committed:
Added missing markdown module. Cleanups.

huwshimi committed:
Abort XHR requests when components are unmounted.

jujugui committed:
Merge pull request #949 from huwshimi/abort-xhrs

Abort XHR

Abort XHR requests when components are unmounted.

jujugui committed:
Merge pull request #948 from huwshimi/remove-subapp

Remove old UI

Remove the subapp, browser, inspector and anything else related to the old UI. I've tried to be as thorough as I can, removing CSS, templates, docs, images, utils and tests.

huwshimi committed:
Show a loading spinner when getting search results.

huwshimi committed:
Update tests for showing the search results spinner.

huwshimi committed:
Fixed lint.

huwshimi committed:
Add Resolve & retry button to the unit errors list.

huwshimi committed:
Added tests for the Resolve & retry button.

huwshimi committed:
Add svg sprites.

huwshimi committed:
Add a component to display the svgs.

huwshimi committed:
Replace inline svgs with svg sprites.

huwshimi committed:
Add changes sprites.

Commit:
Fix tests

jujugui committed:
Merge pull request #912 from makyo/subordinate-uncommitted-unit-count

Add units to subordinate services

This branch ensures that subordinate services' unit lists contain lists of all of the units to which the subordinate has been deployed.

Note that, due to the nature of the ECS, this only applies to deployed units, not uncommitted, until we can figure out how to mark uncommitted relations as scoped to container.

huwshimi committed:
Replace sprites with svg sprites.

huwshimi committed:
Remove old bundle view.

huwshimi committed:
Remove old assets.

huwshimi committed:
Remove the old sprite generation.

huwshimi committed:
Use state to set width and height once.

huwshimi committed:
Handle cancelling the config changes.

huwshimi committed:
Spaces between classes.

huwshimi committed:
Use a map for getting the highest name count.

huwshimi committed:
Added eslint indent rule.

huwshimi committed:
Add no debugger rule.

huwshimi committed:
Remove unused variables from tests.

huwshimi committed:
Remove unused variables.

Commit:
* Missing Wily references

fabricematrat committed:
Bump version: 1.8.999 → 1.8.1000

Commit:
* fix comma

jujugui committed:
Merge pull request #958 from urosj/pyramid-fork

* Missing Wily references

jujugui committed:
Merge pull request #957 from huwshimi/add-eslint-rules

Add eslint rules

Add rules to eslint for `debugger;` and unused variables.

jujugui committed:
Merge pull request #955 from huwshimi/cancel-config

Cancel config

Handle cancelling the config changes.

huwshimi committed:
Fixed new lint rules.

huwshimi committed:
Fixed new lint rules.

huwshimi committed:
Fixed new lint rules.

jujugui committed:
Merge pull request #952 from huwshimi/add-resolve-retry

Add resolve retry

Add Resolve & retry button to the unit errors list.

jujugui committed:
Merge pull request #951 from huwshimi/search-results-spinner

Search results spinner

Show a loading spinner when getting search results.

jujugui committed:
Merge pull request #956 from huwshimi/service-count-map

Service count map

Use a map for getting the highest service name count.

jujugui committed:
Merge pull request #954 from huwshimi/add-svg-sprites

Add svg sprites

Use svg sprites instead of pngs.

huwshimi committed:
Upgrade eslint.

huwshimi committed:
Update switch case rule. Fixed lint in files.

huwshimi committed:
Fixed indentation.

jujugui committed:
Merge pull request #961 from huwshimi/eslint-indents

Eslint indents

Added eslint indentation rule. Fixed indentation.
